src/SFA.DAS.EmployerApprenticeshipsService.Domain/Data/Repositories/IAccountRepository.cs

```diff
@@ -11,7 +11,7 @@ namespace SFA.DAS.EAS.Domain.Data.Repositories
     public interface IAccountRepository
     {
         Task AddPayeToAccount(Paye payeScheme);
-        Task<CreateAccountResult>  CreateAccount(long userId, string employerNumber, string employerName, string employerRegisteredAddress, DateTime? employerDateOfIncorporation, string employerRef, string accessToken, string refreshToken, string companyStatus, string employerRefName, short source, short? publicSectorDataSource, string sector, string aorn);
+        Task<CreateAccountResult> CreateAccount(long userId, string employerNumber, string employerName, string employerRegisteredAddress, DateTime? employerDateOfIncorporation, string employerRef, string accessToken, string refreshToken, string companyStatus, string employerRefName, short source, short? publicSectorDataSource, string sector, string aorn, bool Eoi);
```

Comment:
That is a long list of parameters :) starting to look like maybe a request type could be worth introducing.

Not required change just commenting as I had to scroooooooolll across to see the diff at the end of the line :)

Reply:
Probably better if the repository takes the domain object it’s trying to save. 👍

Reply:
Some calls have been implemented as a request type.

The command calls a number of SP's

Reply:
If I comment out the lines setting the `Eoi` property in `EmployerAccountController` and `EmployerAccountOrchestrator` then all the tests under `SFA.DAS.EmployerAccounts/Tests` still pass? There does not appear to be any coverage of the logic for setting this flag. 

On that point I don't see why we are setting it in the controller; from what I can see it is a value that we need to persist based purely on the whitelist status of the user with no rules for deciding this in controller or orchestrator.

I can see that we do not, necessarily, want to inject `IAuth..Service` into command handlers to have it set there; preferring the handlers to process the request/command that was passed to them. But if feels like something that is outside of any steps we have. We could maybe use `Mediatr` pre-request pipeline to augment the request based on whitelist status.

Another approach might be to inject a command factory into, say, `EmployerAccountOrchestrator`. So, for example, in `EmployerAccountOrchestrator.CreateNewAccount` instead of...

```
                var result = await Mediator.SendAsync(new CreateAccountCommand
                {
                    ExternalUserId = model.UserId,
                    OrganisationType = model.OrganisationType,
                    OrganisationName = model.OrganisationName,
.
.
.
                    Eoi = model.Eoi
```

we have something like : -

```
            try
            {
                var command = commandFactory.GetCreateAccountCommand();

                command.ExternalUserId = .....

```

or

```
            try
            {
                var command = commandFactory.GetCreateAccountCommandForCreateAccountModel(model);

```

and push the authorization checking into that factory.

We could also just have a model -> command mapper.

Wherever we put that logic we should cover it with tests and having the parameter set at the  _top of the chain_ in the controller is just going to me a lot more tests modifying than we need.

                var result = await Mediator.SendAsync(new CreateAccountCommand
                {
                    ExternalUserId = model.UserId,